streams/src/main/java/org/apache/kafka/streams/kstream/internals/AbstractStream.java

```diff
@@ -99,14 +99,14 @@ static <K, VRight, VLeft, VOut> ValueJoinerWithKey<K, VRight, VLeft, VOut> rever
         return (key, value2, value1) -> joiner.apply(key, value1, value2);
     }
 
-    static <K, V, VOut> ValueMapperWithKey<K, V, VOut> withKey(final ValueMapper<V, VOut> valueMapper) {
-        Objects.requireNonNull(valueMapper, "valueMapper cannot be null");
-        return (readOnlyKey, value) -> valueMapper.apply(value);
+    static <K, V, VOut> ValueMapperWithKey<K, V, VOut> withKey(final ValueMapper<V, VOut> mapper) {
+        Objects.requireNonNull(mapper, "mapper cannot be null");
```

Comment:
align parameter name to `mapper` across the board, to avoid `mapper` vs `valueMapper` vs `valueMapperWithKey` (simplifies error handling, and we don't need to worry about the error message, that can just way `mapper` if we only use `mapper` across the board)